java/client/src/main/java/glide/api/commands/servermodules/Json.java

```diff
@@ -1209,6 +1211,248 @@ public static CompletableFuture<Object> toggle(
+     *           Returns the length of the resulting string after appending <code>value</code> to the string at <code>path</code>.<br>
+     *           If multiple paths match, the length of the last updated string is returned.<br>
+     *           If the JSON value at <code>path</code> is not a string of if <code>path</code> doesn't exist, an error is raised.<br>
+     *           If <code>key</code> doesn't exist, an error is raised.
+     *     </ul>
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "{\"a\":\"foo\", \"nested\": {\"a\": \"hello\"}, \"nested2\": {\"a\": 31}}").get();
+     * var res = Json.strappend(client, gs("doc"), gs("baz"), gs("$..a")).get();
+     * assert Arrays.equals((Object[]) res, new Object[] {6L, 8L, null}); // The new length of the string values at path '$..a' in the key stored at `doc` after the append operation.
+     * res = Json.strappend(client, gs("doc"), gs("'\"foo\"'"), gs("nested.a")).get();
+     * assert (Long) res == 11L; // The length of the string value after appending "foo" to the string at path 'nested.array' in the key stored at `doc`.
+     * var getResult = Json.get(client, gs("doc"), gs("$")).get();
+     * assert getResult.equals("[{\"a\":\"foobaz\", \"nested\": {\"a\": \"hellobazfoo\"}, \"nested2\": {\"a\": 31}}]"); // The updated JSON value in the key stored at `doc`.
+     * }</pre>
+     */
+    public static CompletableFuture<Object> strappend(
+        @NonNull BaseClient client, @NonNull GlideString key, @NonNull GlideString value, @NonNull GlideString path) {
+        return executeCommand(
+            client, new ArgsBuilder().add(gs(JSON_STRAPPEND)).add(key).add(path).add(value).toArray());
+    }
+
+    /**
+     * Appends the specified <code>value</code> to the string stored at the root within the JSON document stored at <code>key</code>.
+     *
+     * @param client The client to execute the command.
+     * @param key The key of the JSON document.
+     * @param value The value to append to the string. Must be wrapped with single quotes. For example, to append "foo", pass '"foo"'.
+     * @return
+     *     Returns the length of the resulting string after appending <code>value</code> to the string at the root.<br>
+     *     If the JSON value at root is not a string, an error is raised.<br>
+     *     If <code>key</code> doesn't exist, an error is raised.
+     * @example
+     *     <pre>{@code
+     * Json.set(client, "doc", "$", "'\"foo\"'").get();
+     * var res = Json.strappend(client, "doc", "'\"baz\"'").get();
+     * assert (Long) res == 6L; // The length of the string value after appending "foo" to the string at root in the key stored at `doc`.
+     * var getResult = Json.get(client, "doc").get();
+     * assert getResult.equals("\"foobaz\""); // The updated JSON value in the key stored at `doc`.
+     * }</pre>
+     */
+    public static CompletableFuture<Object> strappend(
```

Comment:
```suggestion
    public static CompletableFuture<Long> strappend(
```

Reply:
Is this really correct?
```assertArrayEquals(new Object[] {6L, 8L, null}, (Object[]) Json.strappend(client, key, "'\"bar\"'", "$..a").get());```

Reply:
What it has no path, it always returns a number, because behaves like `path = "."` (legacy path)

Reply:
OH, I didn't realise this was the overload. My bad.

Reply:
Don't forget to remove `Long` casting in your integration test.